components/footer-icon.js

```diff
@@ -0,0 +1,101 @@
+import { css, html } from "../html-css-utils.js";
+
+class FooterIcon extends HTMLElement {
+  constructor() {
+    super();
+
+    this.validateAttributes();
+    this.render();
+  }
+
+  validateAttributes() {
+    if (!this.hasAttribute("href")) {
+      throw new Error('The "href" attribute is required.');
+    }
+
+    if (!this.hasAttribute("type")) {
+      throw new Error('The "type" attribute is required.');
+    }
+  }
+
+  render() {
+    this.attachShadow({ mode: "open" });
+    this.shadowRoot.innerHTML = this.createCss() + this.createHtml();
+    this.updateIcon();
+  }
+
+  createCss() {
+    return css`
+      .footer-icon {
+        display: inline-block;
+      }
+      slot {
+      display: none;
+    `;
```

Comment:
CSS 작성 형식에 오류가 있는 것 같아요.
slot 부분에서 닫힌 괄호가 없습니다.

```suggestion
    return css`
      .footer-icon {
        display: inline-block;
      }
      
      slot {
      	display: none;
    	}
  	`;
```

Reply:
넵! 확인해서 수정했습니다. 감사합니다.